engine/src/main/java/org/hibernate/validator/spi/property/PropertyAccessorSelector.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.spi.property;
+
+import java.lang.reflect.Method;
+
+/**
+ * Allows to customize property naming, selection and retrieval based on reflection metadata.
+ *
+ * @since 6.1
+ */
+public interface PropertyAccessorSelector {
+
+	/**
+	 * Derives property name from a method. If it is not possible will return {@code null}.
+	 *
+	 * TODO: should this support non-getters methods as well (like setters, void return type etc.) ?
+	 *
+	 * <p>Example for classic java bean convention: {@code getFoo()} -> {@code foo}</p>
+	 *
+	 * @param method method to be analyzed
+	 * @return property name which is exposed by this method, {@code null} if method is unknown to current
+	 * selector instance.
+	 */
+	String getPropertyName(Method method);
```

Comment:
Hi @asereda-gs, thanks a lot for moving forward with this work!

The contract looks ok for Java-based models, but one related future work I'd like us to keep in mind is [HV-1000](https://hibernate.atlassian.net/browse/HV-1000) which is about supporting other models than JavaBeans, e.g. map-based models (with key/value pairs as properties), or JSON-based models etc. This will not let us with `Method` objects, though. Instead we'd have to work with a more abstract representation of "types" and "properties". E.g. something like this:

    public interface PropertySelector {

        Set<Property> getProperties(Type type);
    }

    interface Property {
        String getName();
        Object getValueFrom(Object bean);
    }

    class JavaBeansField implements Property {
        ...
    }

    class JavaBeansGetter implements Property {
        ...
    }

    interface Type { ... }

    /* Implementation based on classes */
    class JavaBean implements Type { ... }

Then, your selector would return `JavaBeansGetter` based on your more relaxed naming pattern.

Do you think you could explore this approach a bit? I realize it's asking a lot, the vision is blurry and it's quite an expansion of what's needed for your very problem. But it'd be great if we could avoid introducing an SPI contract which gets obsolete soon again. Any thoughts?